nnvm/python/nnvm/testing/check_computation.py

```diff
@@ -0,0 +1,451 @@
+                   dump_graph=False):
+    """Compute the function and/or its gradients on a random input and raise
+    an exception if the result doesn't match the reference implementation.
+
+    Parameters
+    ----------
+    symbol : nnvm.Symbol
+        A symbol representing the output.
+
+    grad_input_vars : List[nnvm.Symbol or str or (str, Tuple[int])], optional
+        A list of variables with respect to which the gradients will be computed.
+        None (default) means that all input variables will be used. May be a
+        list of pairs `(var_name, shape)`.
+
+    np_forward : Callable[..., List[numpy.ndarray]], optional
+        A reference implementation to compare with.
+
+    np_backward : Callable[..., List[numpy.ndarray]], optional
+        A reference implementation of gradients. Should also accept head_grads besides
+        normal inputs. Should return gradients with respect to variables from grad_input_vars in
+        exactly the same order.
+
+    shape : Dict[str, Tuple], optional
+        A dict mapping input variable names to shapes.
+        By default shapes will be inferred automatically.
+
+    dtype : Dict[str, str] or str, optional
+        A dict mapping input variable names to dtypes, or just a single dtype.
+        By default dtypes will be inferred automatically.
+
+    in_range : Dict[str, (float, float)] or (float, float), optional
+        A dict mapping input variable names to ranges or just a single range
+        (the same for all variables). Input values will be generated from
+        uniform distributions on these ranges. `head_grads` can also be
+        assigned a range this way.
+
+    exclude_targets : Set[str]
+        Skip compiling and running anything for these targets.
+
+    only_targets : Set[str]
+        Test only for those targets from `ctx_list()` that are also in this set.
+
+    numerical_grads : bool or "if_possible"
+        Whether to additionally check against numerically computed gradients. If 'if_possible' is
+        passed (which is the default) then it will try to create a gradient computation graph and
+        then check gradients numerically only if this graph can be created (i.e. if there are some
+        operations with unimplemented gradients, it will just issue a warning).
+
+    delta : float
+        A small value used for numerical gradient computation (usually called h in textbooks).
+
+    atol : float
+        Absolute tolerance.
+
+    rtol : float
+        Relative tolerance.
+
+    ng_atol : float
+        Absolute tolerance for numerical grad checking.
+
+    ng_rtol : float
+        Relative tolerance for numerical grad checking.
+
+    ng_max_error : float
+        Discard numerical partial derivatives whose estimated error is larger than this value.
+
+    ng_max_discarded_frac : float
+        Allow discarding no more than this fraction of partial derivatives.
+
+    dump_graph : bool
+        Dump the graph even on success.
+    """
```

Comment:
Not sure an option to custom comparison function argument make any sense here. 
Just raising as it came across my thought.